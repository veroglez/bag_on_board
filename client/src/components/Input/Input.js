import React from 'react';
import { useStateValue } from 'common/context/contextHooks';

const onChange = (event, dispatch, { form }) => {
  dispatch({
    type: 'update',
    payload: {
      form: { ...form, name: event.target.value },
    },
  });
};

const Counter = () => {
  const [state, dispatch] = useStateValue();

  return (
    <input
      type="text"
      name="name"
      value={state.value}
      onChange={event => onChange(event, dispatch, state)}
    />
  );
};

export default Counter;
