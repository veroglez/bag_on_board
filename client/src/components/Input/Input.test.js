
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Context from 'common/context/contextHooks';
import Input from './Input';

Enzyme.configure({ adapter: new Adapter() });

const STATE = {
  l10n: { PLUS: '+', MINUS: '-' },
  form: { name: 'Vero' },
};

const HOC = () => (
  <Context.StateContext.Provider value={[STATE]}>
    <Input />
  </Context.StateContext.Provider>
);

describe('<Input>', () => {
  it('render', () => {
    const tree = renderer.create(<HOC />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispath is called with', async () => {
    const dispatch = jest.fn();
    const stateUpdated = {
      type: 'update',
      payload: { form: { name: 'Vero' } },
    };
    const event = { target: { value: 'Vero' } };

    jest
      .spyOn(Context, 'useStateValue')
      .mockImplementation(() => [STATE, dispatch]);
    const wrapper = mount(<HOC />);
    wrapper.find('input').props().onChange(event);

    expect(dispatch).toHaveBeenCalledWith(stateUpdated);
  });
});
