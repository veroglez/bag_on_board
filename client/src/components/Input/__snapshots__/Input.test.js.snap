// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> render 1`] = `
<input
  onChange={[Function]}
  type="text"
/>
`;
